datalake/src/main/java/com/sequenceiq/datalake/service/resize/recovery/ResizeRecoveryService.java

```diff
@@ -0,0 +1,116 @@
+     *
+     * @param crn       the CRN of the user requesting recovery
+     * @param nameOrCrn the name or CRN of the resource to recover
+     * @return true if resize recovery is possible.
+     */
+    public boolean canRecover(String crn, NameOrCrn nameOrCrn) {
+        SdxCluster cluster = sdxService.getByNameOrCrn(crn, nameOrCrn);
+        FlowLog flowLog = flow2Handler.getFirstStateLogfromLatestFlow(cluster.getId());
+        if (DatalakeResizeFlowEventChainFactory.class.getSimpleName().equals(flowChainLogService.getFlowChainType(flowLog.getFlowChainId()))) {
+            SdxStatusEntity actualStatusForSdx = getSdxStatusEntity(crn, nameOrCrn);
+            if (actualStatusForSdx.getStatus().equals(DatalakeStatusEnum.STOP_FAILED)) {
+
+                return true;
+            }
+
+        }
+        return false;
+    }
+
+    private SdxStatusEntity getSdxStatusEntity(String crn, NameOrCrn nameOrCrn) {
+        SdxCluster cluster = sdxService.getByNameOrCrn(crn, nameOrCrn);
+        return sdxStatusService.getActualStatusForSdx(cluster);
+    }
+
+    /**
+     * Determines if it's possible to run a <em>Resize</em> recovery, returning a detailed validation message.
+     * <p>
+     * This is distinguished from {@code canRecover} by including a detailed validation message, usable as a web response.
+     *
+     * @param crn       the CRN of the user requesting recovery
+     * @param nameOrCrn the name or CRN of the resource to recover
+     * @return detailed validation message, as a web response
+     */
+    public SdxRecoverableResponse validateRecovery(String crn, NameOrCrn nameOrCrn) {
+        SdxStatusEntity actualStatusForSdx = getSdxStatusEntity(crn, nameOrCrn);
+        if (actualStatusForSdx.getStatus().equals(DatalakeStatusEnum.STOP_FAILED)) {
+            return new SdxRecoverableResponse("Resize can be recovered from a failed stop", RecoveryStatus.RECOVERABLE);
+        }
+        return new SdxRecoverableResponse("Resize can not be recovered from this point", RecoveryStatus.NON_RECOVERABLE);
+    }
+
+    /**
+     * Checks if recovery is possible, then performs the resize recovery if appropriate.
+     *
+     * @param crn                the CRN of the user requesting recovery
+     * @param nameOrCrn          the name or CRN of the resource to recover
+     * @param sdxRecoveryRequest detailed information about the recovery
+     * @return a response containing information for the triggered recovery Flow.
+     */
+    public SdxRecoveryResponse triggerRecovery(String crn, NameOrCrn nameOrCrn, SdxRecoveryRequest sdxRecoveryRequest) {
+        SdxStatusEntity actualStatusForSdx = getSdxStatusEntity(crn, nameOrCrn);
+        if (actualStatusForSdx.getStatus().equals(DatalakeStatusEnum.STOP_FAILED)) {
+            SdxCluster cluster = sdxService.getByNameOrCrn(crn, nameOrCrn);
+
+            return new SdxRecoveryResponse(sdxReactorFlowManager.triggerSdxStartFlow(cluster));
```

Comment:
I assume we will later tie this to the flow chain I am working on in CB-14399 right? Just making sure.